### PR TITLE
[LF] make generic Dev interpretation error

### PR DESCRIPTION
**Pull request by remyhaemmerle-da**
so we can create such errors at will without breaking canton

sister PR: https://github.com/DACH-NY/canton/pull/13047

### CODE REVIEW

File: daml-lf/transaction/src/main/scala/com/daml/lf/Error.scala [hunk 1]
```diff
@@ -146,69 +146,75 @@ object Error {
 
   final case class ContractIdInContractKey(key: Value) extends Error
 
-  @deprecated("use Limit.ValueNesting", since = "2.0.0")
-  val ValueExceedsMaxNesting: Limit.ValueNesting.type = Limit.ValueNesting
+  // Error that can be thrown by dev or PoC feature only
+  final case class Dev(location: String, error: Dev.Error) extends Error
```
Comment (nickchapman-da) [inline daml-lf/transaction/src/main/scala/com/daml/lf/Error.scala:150]: Nice :+1:

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala [hunk 2]
```diff
@@ -238,25 +239,29 @@ final class Conversions(
...
+                  case Dev.Limit(limitError) =>
+                    limitError match {
+                      case Dev.Limit.ValueNesting(_) =>
+                        builder.setValueExceedsMaxNesting(proto.Empty.newBuilder)
+                      // TODO https://github.com/digital-asset/daml/issues/11691
+                      //   Handle the other cases properly.
+                      case _ =>
+                        builder.setCrash(s"A limit was overpassed when building the transaction")
+                    }
+                  case Dev.ChoiceGuardFailed(coid, templateId, choiceName, byInterface) =>
```
Comment (nickchapman-da) [inline compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:253]: I realise you are preserving existing behaviour.

But going forward... is there any need to add cases for new Dev errors?

**Review by nickchapman-da — APPROVED** [review]
LGTM.
I added just one question.

File: compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala (hunk 2 above)
Comment (remyhaemmerle-da) [reply compiler/scenario-service/server/src/main/scala/com/digitalasset/daml/lf/scenario/Conversions.scala:253]: We want to add new Dev error like the `WronglyTypeSoftFetch` from @moisesackerman-da's #16839

**Review by akrmn — APPROVED** [review]
nice work, thank you @remyhaemmerle-da!

File: build.sh [hunk 3]
```diff
@@ -18,7 +18,7 @@ if [[ "$(uname)" == "Darwin" ]]; then
   tag_filter="$tag_filter,-dont-run-on-darwin,-scaladoc,-pdfdocs"
 fi
 
-SKIP_DEV_CANTON_TESTS=false
+SKIP_DEV_CANTON_TESTS=true
```
Comment (remyhaemmerle-da) [inline build.sh:21]: @garyverhaegen-da is it what I am suppose to do to deactivate test mark with the tag `dev-canton-test` ?

Comment (garyverhaegen-da) [reply build.sh:21]: Yes.

Comment (remyhaemmerle-da) [reply build.sh:21]: cc @moisesackerman-da @nickchapman-da

**Pull request merged**